### PR TITLE
Added option to configure max_header_value_len.

### PR DESCRIPTION
**Pull request by mk46**
Signed-off-by: Manish Kumar <manish.kumar1@india.nec.com>

 
Commit Message: Added option to configure max_header_value_len. 
Additional Description: MAX_HEADER_VALUE_LEN is hardcoded, this PR will make it configurable.
Risk Level:
Testing:
Docs Changes: API
Release Notes: n/a

### CODE REVIEW

**Comment by htuch** [comment]
@fuqianggao this is a nice simple API change to look at.

File: api/envoy/extensions/filters/http/header_to_metadata/v3/header_to_metadata.proto [hunk 1]
```diff
@@ -129,4 +129,7 @@ message Config {
 
   // The list of rules to apply to responses.
   repeated Rule response_rules = 2;
+
+  // Configure max_header_value_len.
```
Comment (markdroth) [inline api/envoy/extensions/filters/http/header_to_metadata/v3/header_to_metadata.proto:133]: This comment isn't very useful.  What does this value actually do?  And what is the behavior if the header is longer than this size?

Comment (mk46) [reply api/envoy/extensions/filters/http/header_to_metadata/v3/header_to_metadata.proto:133]:
```
Configure maximum lenth for :ref:`header value 
<envoy_v3_api_field_extensions.filters.http.header_to_metadata.v3.Config.KeyValuePair.value>` 
to be added in metadata. If this value exceed, no metadata will be added. If not specified, default is 8KiB.
```

Comment (mk46) [reply api/envoy/extensions/filters/http/header_to_metadata/v3/header_to_metadata.proto:133]: Is this make sense?

Comment (markdroth) [reply api/envoy/extensions/filters/http/header_to_metadata/v3/header_to_metadata.proto:133]: That sounds fine.  Please update the comment in the PR.

Comment (mk46) [reply api/envoy/extensions/filters/http/header_to_metadata/v3/header_to_metadata.proto:133]: sure, thanks

File: api/envoy/extensions/filters/http/header_to_metadata/v3/header_to_metadata.proto [hunk 2]
```diff
@@ -129,4 +129,8 @@ message Config {
 
   // The list of rules to apply to responses.
   repeated Rule response_rules = 2;
+
+  // Configures maximum length for :ref:`header value <envoy_v3_api_field_extensions.filters.http.header_to_metadata.v3.Config.KeyValuePair.value>`
```
Comment (fuqianggao) [inline api/envoy/extensions/filters/http/header_to_metadata/v3/header_to_metadata.proto:133]: is the unit for this value Byte? Probably worth mentioning that in the comment.

Comment (mk46) [reply api/envoy/extensions/filters/http/header_to_metadata/v3/header_to_metadata.proto:133]: Yes, "header value" is a string. I don't think it's explicitly required to mention.

**Comment by mk46** [comment]
Hi @zuercher, Would you mind to take a look?

**Comment by mk46** [comment]
Pinging for not to stale.

**Pull request closed without merging**